catalogue/webapp/pages/works.js

```diff
@@ -57,72 +63,149 @@ const SearchWorks = ({
+                  {pagination && pagination.range &&
+                    <Fragment>
+                      <div className={`flex flex--v-center font-pewter ${font({s: 'LR3', m: 'LR2'})}`}>
+                        Showing {pagination.range.beginning} - {pagination.range.end}
+                      </div>
+                      <Pagination
+                        prevPage={pagination.prevPage}
+                        currentPage={pagination.currentPage}
+                        pageCount={pagination.pageCount}
+                        nextPage={pagination.nextPage}
+                        nextQueryString={pagination.nextQueryString}
+                        prevQueryString={pagination.prevQueryString} />
+                    </Fragment>
+                  }
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className={`row ${spacing({s: 4}, {padding: ['top']})}`}>
+          <div className="container">
+            <div className="grid">
+              {works.results.map(result => (
+                <div key={result.id} className={grid({s: 6, m: 4, l: 3, xl: 2})}>
+                  <Promo
+                    sizes='(min-width: 1340px) 178px, (min-width: 960px) calc(25vw - 52px), (min-width: 600px) calc(33.24vw - 43px), calc(50vw - 27px)'
+                    id={result.id}
+                    contentType='work'
+                    image={{
+                      contentUrl: result.thumbnail && result.thumbnail.url,
+                      width: 300,
+                      height: 300,
+                      alt: ''
+                    }}
+                    datePublished={result.createdDate && result.createdDate.label}
+                    isConstrained={true}
+                    title={result.title}
+                    defaultSize={180}
+                    url={`/works/${result.id}${getQueryParamsForWork(query)}`} />
+                </div>
+              ))}
+            </div>
+          </div>
+        </div>
+      </Fragment>
+    }
   </DefaultPageLayout>
 );
 
-const WorksPage = withFormik({
-  mapPropsToValues: props => props, // this passed the initial props from `getInitialProps down`
-  handleSubmit: async ({query, works}, actions) => {
-    const res = await fetch(`https://api.wellcomecollection.org/catalogue/v1/works?query=${query}`);
-    const json = await res.json();
-    actions.setValues({ works: json, query });
-    actions.setSubmitting(false);
+class Works extends Component<Props> {
+  handleSubmit: Function;
+  static getInitialProps: Function;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event: any) {
+    event.preventDefault();
+
+    const queryString = encodeURIComponent(event.target[0].value);
+
+    // Update the URL, which in turn will update props
+    Router.push({
+      pathname: '/works',
+      query: {query: queryString, page: '1'}
```

Comment:
`page` should be dynamic no?
Also, I wonder if we should be doing state management, this seems like we're hacking the router to deal with our state, and I imagine there might be performance benefits we're losing out on.